compiler-plugin-tests/src/test/java/io/ballerina/stdlib/kafka/plugin/CodeSnippetGenerationCodeActionTest.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.stdlib.kafka.plugin;
+
+import io.ballerina.projects.plugins.codeaction.CodeActionArgument;
+import io.ballerina.projects.plugins.codeaction.CodeActionInfo;
+import io.ballerina.tools.text.LinePosition;
+import io.ballerina.tools.text.LineRange;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+
+import static io.ballerina.stdlib.kafka.plugin.KafkaCodeTemplate.NODE_LOCATION;
+
+/**
+ * A class for testing code actions.
+ */
+public class CodeSnippetGenerationCodeActionTest extends AbstractCodeActionTest {
+
+    @Test(dataProvider = "testDataProvider")
+    public void testEmptyServiceCodeAction(String srcFile, int line, int offset, String resultFile)
+            throws IOException {
+        Path filePath = RESOURCE_PATH.resolve("ballerina_sources")
+                .resolve("snippet_gen_service_1")
+                .resolve(srcFile);
+        Path resultPath = RESOURCE_PATH.resolve("expected_sources")
+                .resolve("service_1")
+                .resolve(resultFile);
+        performTest(filePath, LinePosition.from(line, offset),
+                getExpectedCodeAction("service.bal", 2, 64), resultPath);
+    }
+
+    @Test(dataProvider = "testDataProvider")
+    public void testServiceWithVariablesCodeAction(String srcFile, int line, int offset, String resultFile)
+            throws IOException {
+        Path filePath = RESOURCE_PATH.resolve("ballerina_sources")
+                .resolve("snippet_gen_service_2")
+                .resolve(srcFile);
+        Path resultPath = RESOURCE_PATH.resolve("expected_sources")
+                .resolve("service_2")
+                .resolve(resultFile);
+        performTest(filePath, LinePosition.from(line, offset),
+                getExpectedCodeAction("service.bal", 5, 1), resultPath);
+    }
+
+    @DataProvider
+    private Object[][] testDataProvider() {
+        return new Object[][]{
+                {"service.bal", 2, 0, "result.bal"}
+        };
+    }
```

Comment:
If we are writing separate test cases for testing empty services and services with variables, I think there is no need of introducing a `@DataProvider`. We can directly pass the parameters to `performTest` inside the test itself. It will be more readable as there is no gain in introducing a `@DataProvider` for a scenario like this.

Reply:
Added in https://github.com/ballerina-platform/module-ballerinax-kafka/pull/362/commits/87bf428c9360568a8df0f416575540893d7e1792